### ISSUE TITLE
[Doc] How to write custom propagator

### ISSUE THREAD

**Issue by cijothomas** [open-telemetry/opentelemetry-dotnet#3028]
No doc exists today on this topic. Opening an issue to track adding this one.
Need to decide if it should be under tracing or independent of tracing.
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/extending-the-sdk

**Comment by TimothyMothra** [comment]
@cijothomas, is this still needed?

**Comment by cijothomas** [comment]
yes but it's more involved now due to the fact that runtime also have the api for propagation and we need to eventually deprecated ours
